File: lab06/all_whales.pl
#!/usr/bin/perl -w

#$whale = $ARGV[0];
#$nPods = 0;
#$nIndv = 0;

while ($line = <STDIN>) {
   chomp $line;
   $whale = $line;
   $whale =~ tr/A-Z/a-z/;
   $whale =~ s/[0-9]+ +//;
   $whale =~ s/s? *$//;
   $whale =~ s/ +/ /g;
   $nInPod = $line;
   $nInPod =~ s/ .*//;

   $nPods{$whale}++;
   $nIndv{$whale} += $nInPod;
}

foreach $whale (sort keys %nPods) {
   print "$whale observations: $nPods{$whale} pods, $nIndv{$whale} individuals\n"
}

File: lab06/all_whales.py
#!/usr/bin/env python3

import sys, re

nPods = {}
nIndv = {}

while 1:
   line = sys.stdin.readline()
   if not line:
      break

   whale = str.lower(line)
   whale = re.sub('^\d+ +', '', whale)
   whale = re.sub('s?\s*$', '', whale)
   whale = re.sub(' +', ' ', whale)
   nInPod = int(re.sub(' .*', '', line))

   nPods[whale]= nPods.get(whale,0) + 1

   try:
      nIndv[whale] += nInPod
   except:
      nIndv[whale] = nInPod

for whale in sorted(nPods.keys()):
   print("%s observations: %d pods, %d individuals" % (whale, nPods[whale], nIndv[whale]))
